tests/phpunit/tests/general/feedLinksExtra.php

```diff
@@ -510,4 +510,77 @@ static function() {
 		$expected .= ' href="http://example.org/?feed=foo&#038;p=' . self::$post_with_comment_id . '" />' . "\n";
 		$this->assertSame( $expected, get_echo( 'feed_links_extra' ) );
 	}
+
+	/**
+	 * @ticket 54703
+	 */
+	public function test_feed_links_extra_should_output_nothing_when_show_comments_feed_filter_returns_false() {
+		add_filter( 'feed_links_show_comments_feed', '__return_false' );
+
+		$this->go_to( get_the_permalink( self::$post_with_comment_id ) );
+		$this->assertEmpty( get_echo( 'feed_links_extra' ) );
+	}
+
+	/**
+	 * @dataProvider data_feed_links_extra_should_output_nothing_when_post_comments_feed_link_is_falsy
+	 *
+	 * @ticket 54703
+	 *
+	 * @param string $callback The callback to use for the 'post_comments_feed_link' filter.
+	 */
+	public function test_feed_links_extra_should_output_nothing_when_post_comments_feed_link_is_falsy( $callback ) {
+		if ( ! str_starts_with( $callback, '__return' ) ) {
+			$callback = array( $this, $callback );
+		}
+
+		add_filter( 'post_comments_feed_link', $callback );
+
+		$this->go_to( get_the_permalink( self::$post_with_comment_id ) );
+		$this->assertEmpty( get_echo( 'feed_links_extra' ) );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_feed_links_extra_should_output_nothing_when_post_comments_feed_link_is_falsy() {
+		return array(
+			'empty string' => array( 'callback' => '__return_empty_string' ),
+			'empty array'  => array( 'callback' => '__return_empty_array' ),
+			'zero int'     => array( 'callback' => '__return_zero' ),
+			'zero float'   => array( 'callback' => 'cb_return_zero_float' ),
+			'zero string'  => array( 'callback' => 'cb_return_zero_string' ),
```

Comment:
```suggestion
			'zero float'   => array( 'callback' => [ $this, 'cb_return_zero_float' ] ),
			'zero string'  => array( 'callback' => [ $this, 'cb_return_zero_string' ] ),
```

Is there a reason you can't do this (but per the coding standards) to avoid the conditional in the test?